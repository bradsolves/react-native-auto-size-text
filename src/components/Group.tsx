import React, { useState } from 'react'
import { Text } from 'react-native'

import { AutoSizeTextProps } from '../types'

const Group = (props: AutoSizeTextProps) => {
  const [currentFont] = useState(2048)

  const { children, style } = props

  return (
    <Text
      adjustsFontSizeToFit
      style={[
        style,
        {
          fontSize: currentFont,
        },
      ]}
    >
      {children}
    </Text>
  )
}

export default Group
