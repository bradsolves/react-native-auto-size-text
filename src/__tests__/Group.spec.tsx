import { cleanup, render, waitFor } from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Group from '../components/Group';

afterEach(cleanup);

describe('Group snapshot', () => {
  it('MaxLines renders correctly', () => {
    const tree = renderer.create(<Group>Lorem impsum</Group>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Group mode', () => {
  it('should pass the right props to Text component inside Group', async () => {
    const mockContent = 'Lorem ipsum dolor sit amet.';
    const mockColor = 'red';

    const { getByTestId } = render(
      <Group style={{ color: mockColor }}>{mockContent}</Group>
    );

    const textComponent = getByTestId('group');
    const content = textComponent.props.children;
    const adjustsFontSizeToFit = textComponent.props.adjustsFontSizeToFit;
    const color = textComponent.props.style[0].color;

    await waitFor(() => {
      expect(content).toBe(mockContent);
      expect(color).toBe(mockColor);
      expect(adjustsFontSizeToFit).toBeTruthy();
    });
  });
});
