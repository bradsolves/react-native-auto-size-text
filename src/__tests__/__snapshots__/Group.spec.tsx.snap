// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group snapshot MaxLines renders correctly 1`] = `
<Text
  adjustsFontSizeToFit={true}
  numberOfLines={1000}
  style={
    Array [
      undefined,
      Object {
        "fontSize": 1000,
      },
    ]
  }
  testID="group"
>
  Lorem impsum
</Text>
`;
